chapter-05/2_represent/money/number.go

```diff
@@ -0,0 +1,59 @@
+package money
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Number can represent money with a fixed precision.
+// example: 1.52 = 1 + 52 * 10^(-2) will be stored as {1, 52, 2}
+type Number struct {
+	// the integer part of the Number
+	integerPart int
+	// the decimal part of the Number
+	decimalPart int
+	// precision of the decimal part, as the exponent of a power of 10
+	precision int
+}
+
+const (
+	// ErrInvalidInteger is returned if the integer part is not a number.
+	ErrInvalidInteger = moneyError("unable to convert integer part")
+
+	// ErrInvalidDecimal is returned if the decimal part is not a number.
+	ErrInvalidDecimal = moneyError("unable to convert decimal part")
+
+	// ErrTooLarge is returned if the amount is too large - this would cause floating point precision errors.
+	ErrTooLarge = moneyError("amount over 10^15 is too large")
```

Comment:
different from maxAmount.

`ErrTooLarge = moneyError(fmt.Sprintf("amount over %d is too large", maxAmount)`